cmake/Modules/FindTPLROCTHRUST.cmake

```diff
@@ -12,4 +12,4 @@ FIND_PACKAGE(rocthrust REQUIRED)
 KOKKOS_CREATE_IMPORTED_TPL(ROCTHRUST INTERFACE LINK_LIBRARIES roc::rocthrust)
 
 # Export ROCTHRUST as a Kokkos dependency
-KOKKOS_EXPORT_CMAKE_TPL(ROCTHRUST)
+KOKKOS_EXPORT_CMAKE_TPL(rocthrust)
```

Comment:
I don't understand this change. We should export the TPL that we import. I tested the current code with ArborX when I created the PR and it's working fine.

Reply:
This change also works with ArborX. Are we supposed to export the TPL that we created or the underlying library?

Reply:
I noticed the problem first when working on https://github.com/kokkos/kokkos/pull/6870. If I used `KOKKOS_EXPORT_CMAKE_TPL` with the new TPL created by `Kokkos`, `find_dependency` didn't have a hint where to find the underlying library the interface target tried to link with. Basically, `find_dependency` here would call `find_package(ROCTHRUST)` with the `ROCTHRUST_DIR` set (which doesn't exist since it's an interface target). These changes should only make a difference if the `ROCm` installation doesn't define the `roc::rocthrust` target already and you need to provide hints. Also, we should try to find `rocprim` and link to it, see https://github.com/ROCm/rocThrust?tab=readme-ov-file#using-rocthrust-in-a-project.